packages/playwright-core/src/server/frameSelectors.ts

```diff
@@ -75,11 +75,12 @@ export class FrameSelectors {
     }, { info: resolved.info, scope: resolved.scope });
   }
 
-  async queryCount(selector: string): Promise<number> {
+  async queryCount(selector: string, options: any): Promise<number> {
     const resolved = await this.resolveInjectedForSelector(selector);
     // Be careful, |this.frame| can be different from |resolved.frame|.
     if (!resolved)
       throw new Error(`Failed to find frame for selector "${selector}"`);
+    await options.__testHookAfterStable?.();
```

Comment:
I find the current hook name unsuitable.

```suggestion
    await options.__testHookBeforeQuery?.();
```